tests/rptest/services/admin.py

```diff
@@ -157,12 +157,14 @@ def _get_stable_configuration(
                     f"get status:{meta['status']} while already observed:{status} before"
                 )
                 return None
-            read_replicas = set([r['node_id'] for r in meta["replicas"]])
+            read_replicas = meta["replicas"]
```

Comment:
you are changing the type of PartitionDetails::replicas from set of ints to probably dicts. have you inspected all the uses of this API to make sure its safe?

before it was easy to understand the type of `replicas` just by looking at admin.py but now a reader should go to the redpanda internals to deduce it, it's inconvenient, please introduce new type e.g. Replica with  node_id and core fields to make the type clear